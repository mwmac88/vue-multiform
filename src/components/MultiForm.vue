<template>
  <v-container>
    <v-layout text-xs-center wrap>
      <v-flex xs12>
        <v-card class="elevation-12">
          <v-stepper v-model="activeStep">
            <v-stepper-header>
              <v-stepper-step
                v-for="(step, index) in steps"
                :key="index"
                :editable="index + 1 < activeStep"
                :complete="index + 1 < activeStep"
                :step="index + 1"
              >{{ step }}</v-stepper-step>
            </v-stepper-header>

            <v-stepper-items>
              <v-stepper-content step="1">
                <StepPersonal @nextStep="nextStep"/>
              </v-stepper-content>

              <v-stepper-content step="2">
                <StepFiles @nextStep="nextStep" @previousStep="previousStep"/>
              </v-stepper-content>

              <v-stepper-content step="3">
                <StepAbout @formComplete="sendForm" @previousStep="previousStep"/>
              </v-stepper-content>
            </v-stepper-items>
          </v-stepper>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import StepPersonal from "./FormSteps/StepPersonal";
import StepFiles from "./FormSteps/StepFiles";
import StepAbout from "./FormSteps/StepAbout";

export default {
  name: "multi-form",
  components: {
    StepPersonal,
    StepFiles,
    StepAbout
  },
  data: () => ({
    activeStep: 1,
    steps: ["Personal", "Files", "About"]
  }),
  methods: {
    nextStep() {
      this.activeStep++;
    },
    previousStep() {
      this.activeStep--;
    },
    sendForm() {
        
    }
  }
};
</script>

<style>
</style>
