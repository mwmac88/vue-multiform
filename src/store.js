import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

const formData = new FormData();

export default new Vuex.Store({
  state: {
    formData: formData,
  },
  mutations: {
    updateField(state, args) {
      const [ field, value ] = [...args];
      state[field] = value;
    }
  },
  actions: {
    triggerUpdateField({ commit }, args) {
      commit('updateField', args);
    }
  },
});
