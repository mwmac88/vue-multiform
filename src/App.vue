<template>
  <v-app>
    <v-content>
      <MultiForm/>
    </v-content>
  </v-app>
</template>

<script>
import MultiForm from "./components/MultiForm";

export default {
  name: "app",
  components: {
    MultiForm
  },
};
</script>
