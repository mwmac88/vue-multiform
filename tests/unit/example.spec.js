import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import MultiForm from '@/components/MultiForm.vue';
import { getters } from '../../src/store';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('MultiForm.vue', () => {
  let getters;
  let store;

  beforeEach(() => {
    getters = {
      getInputKeys: jest.fn(),
    };
    store = new Vuex.Store({
      getters,
    });
  });

  it('renders vue component successfully', () => {
    const wrapper = shallowMount(MultiForm, { store, localVue });
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
