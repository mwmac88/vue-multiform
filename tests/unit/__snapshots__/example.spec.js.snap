// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiForm.vue renders vue component successfully 1`] = `
<v-container>
  <v-layout
    text-xs-center=""
    wrap=""
  >
    <v-flex
      xs12=""
    >
      <v-card
        class="elevation-12"
      >
        <v-stepper>
          <v-stepper-header />
           
          <v-stepper-items>
            <v-stepper-content
              step="1"
            >
              <steppersonal-stub />
            </v-stepper-content>
             
            <v-stepper-content
              step="2"
            >
              <stepfiles-stub />
            </v-stepper-content>
             
            <v-stepper-content
              step="3"
            >
              <stepabout-stub />
            </v-stepper-content>
          </v-stepper-items>
        </v-stepper>
      </v-card>
    </v-flex>
  </v-layout>
</v-container>
`;
